oci/private/pull.bzl

```diff
@@ -361,28 +366,86 @@ def _oci_pull_impl(rctx):
     manifest, size, digest = downloader.download_manifest(rctx.attr.identifier, "manifest.json")
 
     if manifest["mediaType"] in _SUPPORTED_MEDIA_TYPES["manifest"]:
-        if rctx.attr.platform:
-            fail("{}/{} is a single-architecture image, so attribute 'platforms' should not be set.".format(rctx.attr.registry, rctx.attr.repository))
-
+        # plain image manifest: use the contents.
+        pass
     elif manifest["mediaType"] in _SUPPORTED_MEDIA_TYPES["index"]:
+        # image index manifest: download the image manifest for the target platform.
         if not rctx.attr.platform:
-            fail("{}/{} is a multi-architecture image, so attribute 'platforms' is required.".format(rctx.attr.registry, rctx.attr.repository))
+            fail("{}/{} is a multi-architecture image, so attribute 'platforms' is required.".format(
+                rctx.attr.registry,
+                rctx.attr.repository,
+            ))
 
         matching_manifest = _find_platform_manifest(manifest, rctx.attr.platform)
         if not matching_manifest:
-            fail("No matching manifest found in image {}/{} for platform {}".format(rctx.attr.registry, rctx.attr.repository, rctx.attr.platform))
-
-        # extra download to get the manifest for the target platform
+            fail("No matching manifest found in image {}/{} for platform {}".format(
+                rctx.attr.registry,
+                rctx.attr.repository,
+                rctx.attr.platform,
+            ))
         manifest, size, digest = downloader.download_manifest(matching_manifest["digest"], "manifest.json")
     else:
         fail("Unrecognized mediaType {} in manifest file".format(manifest["mediaType"]))
-    
-    # symlink manifest.json to blobs with it's digest. 
+
+    # symlink manifest.json to blobs with its digest.
     # it is okay to use symlink here as copy_to_directory will dereference it when creating the TreeArtifact.
     rctx.symlink("manifest.json", _digest_into_blob_path(digest))
 
-    # download the image config
-    downloader.download_blob(manifest["config"]["digest"], _digest_into_blob_path(manifest["config"]["digest"]))
+    # if the user provided a platform for the image, validate it matches the config as best effort.
+    if rctx.attr.platform:
```

Comment:
Note that this change also affects multi-arch images: we now validate the platform in the images' config metadata (not just in the image index). I don't think this should be considered backward-incompatible because:
- If the os/arch mis-match, there's something very dodgy happening with the user's image index manifest.
- If the os/arch are unset, we don't fail.